pkg/controller/node.go

```diff
@@ -82,13 +83,13 @@ func (r *ReconcileNode) Reconcile(ctx context.Context, request reconcile.Request
 	}
 
 	// If the node has no DeletionTimestamp, it means it's either create event or update event
-	if n.ObjectMeta.DeletionTimestamp.IsZero() && !n.Spec.Unschedulable {
+	if n.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
How about adding more checks here, like below:
if n.ObjectMeta.DeletionTimestamp.IsZero() && !n.Spec.Unschedulable && LabelInstanceTypeStable_is_added _to_node == true {
Because only when schedulable and label_added, addNodeToAvailableAttachmentsMap and tryCreateFailedReplicas can pass.

Reply:
We detect if the label is added from Update predicate, I don't think we need to check `LabelInstanceTypeStable_is_added _to_node == true` here.

Reply:
I am not sure we should tie node "schedulablility" and capacity tracking. I think we should add the node to the map if the deletion timestamp is not set and if the label is added (latter is managed by the predicate). We could, however, check if it is schedulable before calling `tryCreateFailedReplicas`.

Reply:
If we don't add the check `LabelInstanceTypeStable_is_added _to_node == true`, when `schedulable==true` and `LabelInstanceTypeStable_is_added _to_node == false`, `addNodeToAvailableAttachmentsMap()` will be called which is unnecessary(because it will fail for sure).

Reply:
Hi @landreasyan if LabelInstanceTypeStable_is_added _to_node == false, addNodeToAvailableAttachmentsMap will fail at this line:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/7d15002e62d9ab7ea8f0fc43b4396b4528e36749/pkg/azureutils/azure_disk_utils.go#L222

Reply:
I am wondering if we should even return true in the reconciler if the label is not set yet?

Reply:
I think it makes more sense to check the label inside `addNodeToAvailableAttachmentsMap()`. In `addNodeToAvailableAttachmentsMap()`, we check the node capacity from the cache first then label, so we shouldn't stop it just by if label exists.

Reply:
> I am wondering if we should even return true in the predicate if the label is not set yet?

but we want to catch the label update to trigger conciliation so that the node can be added to the map, right?

Reply:
Are there any concerns with the predicate change I suggested below? Should we just go that route?

Reply:
After discussion, we agree to keep predicate check as current iteration and add `!n.Spec.Unschedulable` before calling `tryCreateFailedReplicas`. We could optimize it in the future.